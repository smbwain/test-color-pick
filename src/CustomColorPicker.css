
.custom-color-picker {
    max-width: 300px;
    border: gray 1px solid;
    border-radius: 3px;
    padding: 3px;
    position: relative;
    background-color: #fff;
}

.custom-color-picker input {
    width: 100%;
    border: none;
    outline: none;
}

.custom-color-picker ul {
    border: solid 1px gray;
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
    max-height: 300px;
    overflow-y: auto;
    position: absolute;
    left: -1px;
    right: -1px;
    background-color: white;
    list-style-type: none;
    padding: 0;
    margin: 0;
}

.custom-color-picker ul li {
    padding: 3px;
    text-align: left;
}

.custom-color-picker ul li div {
    width: 10px;
    height: 10px;
    display: inline-block;
    border: black 1px solid;
    padding: 3px;
}