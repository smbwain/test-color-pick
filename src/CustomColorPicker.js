import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './CustomColorPicker.css';

const Highlight = ({str, substr}) => {
    const pos = str.indexOf(substr);
    if (pos === -1) {
        return str;
    }
    return (
        <span>
            {str.substr(0, pos)}
            <b>
                {str.substr(pos, substr.length)}
            </b>
            {str.substr(pos+substr.length)}
        </span>
    );
};

class CustomColorPicker extends Component {
    constructor(props) {
        super(props);
        this.state = {
            value: '',
            selectedIndex: -1,
            showSuggestionList: false,
            suggestedColors: [],
        };
        this.options = {};
        this.inputKeyDown = this.inputKeyDown.bind(this);
        this.inputChange = this.inputChange.bind(this);
    }
    selectColor(color) {
        this.setState({
            value: color.name,
            showSuggestionList: false,
            suggestedColors: this.calcSuggestions(color.name),
            selectedIndex: -1,
        }, () => {
            this.input && this.input.focus();
            if (this.props.onChange) {
                this.props.onChange(color);
            }
        });
    }
    calcSuggestions(value) {
        return this.props.colors.filter(color => color.name.indexOf(value) !== -1);
    }
    makeOptionVisibleScroll(top) {
        const color = this.state.suggestedColors[this.state.selectedIndex];
        const option = color && this.options[color.name];
        if (option && this.optionsContainer && (
            (option.offsetTop + option.offsetHeight > this.optionsContainer.scrollTop + this.optionsContainer.clientHeight) ||
            (option.offsetTop < this.optionsContainer.scrollTop)
        )) {
            option.scrollIntoView(top);
        }
    }
    inputChange(event) {
        const newValue = event.target.value;
        this.setState({
            value: newValue,
            showSuggestionList: true,
            suggestedColors: this.calcSuggestions(newValue),
            selectedIndex: -1,
        });
        if (this.props.onChange) {
            this.props.onChange( this.props.colors.find(color => color.name === newValue) );
        }
    }
    inputKeyDown(event) {
        const {selectedIndex, suggestedColors} = this.state;
        if (event.key === 'ArrowDown') {
            if (selectedIndex < suggestedColors.length-1) {
                this.setState({
                    selectedIndex: selectedIndex + 1,
                }, () => {
                    this.makeOptionVisibleScroll(false);
                });
            }
        } else if (event.key === 'ArrowUp') {
            if (selectedIndex > 0) {
                this.setState({
                    selectedIndex: selectedIndex - 1,
                }, () => {
                    this.makeOptionVisibleScroll(true);
                });
            }
        } else if (event.key === 'Enter' && selectedIndex !== -1) {
            this.selectColor(suggestedColors[selectedIndex]);
        } else {
            return;
        }
        event.preventDefault();
        event.stopPropagation();
    }
    render() {
        const {value, selectedIndex, showSuggestionList, suggestedColors} = this.state;
        return (
            <div className="custom-color-picker">
                <input
                    value={value}
                    onChange={this.inputChange}
                    onKeyDown={this.inputKeyDown}
                    onFocus={() => {
                        this.setState({
                            showSuggestionList: true,
                        });
                    }}
                    onBlur={() => {
                        this.setState({
                            showSuggestionList: false,
                        });
                    }}
                    ref={input => this.input = input}
                />
                { showSuggestionList && !!suggestedColors.length && value.length > 1 && (
                    <ul
                        className="suggestions-list"
                        onMouseLeave={() => {
                            this.setState({
                                selectedIndex: -1,
                            });
                        }}
                        ref={el => {
                            this.optionsContainer = el;
                        }}
                    >
                        { suggestedColors.map((color, i) => (
                            <li
                                key={color.name}
                                style={{
                                    backgroundColor: i === selectedIndex ? '#cacaca' : 'white'
                                }}
                                onMouseDown={() => {
                                    this.selectColor(color);
                                }}
                                onMouseEnter={() => {
                                    this.setState({
                                        selectedIndex: i,
                                    });
                                }}
                                ref={option => {
                                    this.options[color.name] = option;
                                }}
                            >
                                <div
                                    style={{
                                        backgroundColor: `#${color.hex}`,
                                    }}
                                />
                                <Highlight str={color.name} substr={value}/>
                            </li>
                        )) }
                    </ul>
                ) }
            </div>
        );
    }
}

CustomColorPicker.propTypes = {
    onChange: PropTypes.func,
    colors: PropTypes.arrayOf(PropTypes.shape({
        hex: PropTypes.string,
        name: PropTypes.string,
    })).isRequired
};

export default CustomColorPicker;
