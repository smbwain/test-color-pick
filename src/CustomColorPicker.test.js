import React from 'react';
import CustomColorPicker from './CustomColorPicker';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('color-picker', () => {

    let colorPicker;
    let currentColor = undefined;

    it('renders without crashing', () => {
        const colors = [{
            name: 'black',
            hex: '#000',
        }, {
            name: 'brown',
            hex: '#5f4611',
        }, {
            name: 'red',
            hex: '#f00',
        }, {
            name: 'lightred',
            hex: '#f96063',
        }, {
            name: 'white',
            hex: '#fff',
        }];

        const onChange = (newColor) => {
            currentColor = newColor;
        };
        colorPicker = mount(<CustomColorPicker colors={colors} onChange={onChange}/>);

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });

    it('works after 1st symbol typed', () => {
        colorPicker.find('input').simulate('change', {
            target: {
                value: 'r',
            },
        });

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });

    it('shows list 2nd symbol typed', () => {
        colorPicker.find('input').simulate('change', {
            target: {
                value: 're',
            },
        });

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });

    it('highlight option after mouse entered', () => {
        colorPicker.find('li').last().simulate('mouseenter');

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });

    it('hide option highlighting after mouse left', () => {
        colorPicker.find('ul').simulate('mouseleave');

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });

    it('selects color after click on suggested option', () => {
        colorPicker.find('li').last().simulate('mousedown');
        colorPicker.find('input').simulate('blur');

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });

    it('deselects color after start writing', () => {
        colorPicker.find('input').simulate('change', {
            target: {
                value: 're',
            },
        });

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });

    it('selects color after true name of the color was typed', () => {
        colorPicker.find('input').simulate('change', {
            target: {
                value: 'red',
            },
        });

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });

    it('highlights color after click on arrowdown key', () => {
        colorPicker.find('input').simulate('change', {
            target: {
                value: 're',
            },
        });
        colorPicker.find('input').simulate('keydown', {
            key: 'ArrowDown',
        });

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });

    it('selects color after click on enter key', () => {
        colorPicker.find('input').simulate('keydown', {
            key: 'Enter',
        });

        expect(colorPicker.state()).toMatchSnapshot();
        expect(toJson(colorPicker)).toMatchSnapshot();
        expect(currentColor).toMatchSnapshot();
    });
});