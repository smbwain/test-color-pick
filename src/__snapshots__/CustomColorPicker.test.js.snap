// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color-picker deselects color after start writing 1`] = `
Object {
  "selectedIndex": -1,
  "showSuggestionList": true,
  "suggestedColors": Array [
    Object {
      "hex": "#f00",
      "name": "red",
    },
    Object {
      "hex": "#f96063",
      "name": "lightred",
    },
  ],
  "value": "re",
}
`;

exports[`color-picker deselects color after start writing 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="re"
    />
    <ul
      className="suggestions-list"
      onMouseLeave={[Function]}
    >
      <li
        key="red"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f00",
            }
          }
        />
        <Highlight
          str="red"
          substr="re"
        >
          <span>
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
      <li
        key="lightred"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f96063",
            }
          }
        />
        <Highlight
          str="lightred"
          substr="re"
        >
          <span>
            light
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
    </ul>
  </div>
</CustomColorPicker>
`;

exports[`color-picker deselects color after start writing 3`] = `undefined`;

exports[`color-picker hide option highlighting after mouse left 1`] = `
Object {
  "selectedIndex": -1,
  "showSuggestionList": true,
  "suggestedColors": Array [
    Object {
      "hex": "#f00",
      "name": "red",
    },
    Object {
      "hex": "#f96063",
      "name": "lightred",
    },
  ],
  "value": "re",
}
`;

exports[`color-picker hide option highlighting after mouse left 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="re"
    />
    <ul
      className="suggestions-list"
      onMouseLeave={[Function]}
    >
      <li
        key="red"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f00",
            }
          }
        />
        <Highlight
          str="red"
          substr="re"
        >
          <span>
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
      <li
        key="lightred"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f96063",
            }
          }
        />
        <Highlight
          str="lightred"
          substr="re"
        >
          <span>
            light
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
    </ul>
  </div>
</CustomColorPicker>
`;

exports[`color-picker hide option highlighting after mouse left 3`] = `undefined`;

exports[`color-picker highlight option after mouse entered 1`] = `
Object {
  "selectedIndex": 1,
  "showSuggestionList": true,
  "suggestedColors": Array [
    Object {
      "hex": "#f00",
      "name": "red",
    },
    Object {
      "hex": "#f96063",
      "name": "lightred",
    },
  ],
  "value": "re",
}
`;

exports[`color-picker highlight option after mouse entered 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="re"
    />
    <ul
      className="suggestions-list"
      onMouseLeave={[Function]}
    >
      <li
        key="red"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f00",
            }
          }
        />
        <Highlight
          str="red"
          substr="re"
        >
          <span>
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
      <li
        key="lightred"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "#cacaca",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f96063",
            }
          }
        />
        <Highlight
          str="lightred"
          substr="re"
        >
          <span>
            light
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
    </ul>
  </div>
</CustomColorPicker>
`;

exports[`color-picker highlight option after mouse entered 3`] = `undefined`;

exports[`color-picker highlights color after click on arrowdown key 1`] = `
Object {
  "selectedIndex": 0,
  "showSuggestionList": true,
  "suggestedColors": Array [
    Object {
      "hex": "#f00",
      "name": "red",
    },
    Object {
      "hex": "#f96063",
      "name": "lightred",
    },
  ],
  "value": "re",
}
`;

exports[`color-picker highlights color after click on arrowdown key 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="re"
    />
    <ul
      className="suggestions-list"
      onMouseLeave={[Function]}
    >
      <li
        key="red"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "#cacaca",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f00",
            }
          }
        />
        <Highlight
          str="red"
          substr="re"
        >
          <span>
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
      <li
        key="lightred"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f96063",
            }
          }
        />
        <Highlight
          str="lightred"
          substr="re"
        >
          <span>
            light
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
    </ul>
  </div>
</CustomColorPicker>
`;

exports[`color-picker highlights color after click on arrowdown key 3`] = `undefined`;

exports[`color-picker renders without crashing 1`] = `
Object {
  "selectedIndex": -1,
  "showSuggestionList": false,
  "suggestedColors": Array [],
  "value": "",
}
`;

exports[`color-picker renders without crashing 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value=""
    />
  </div>
</CustomColorPicker>
`;

exports[`color-picker renders without crashing 3`] = `undefined`;

exports[`color-picker selects color after click on enter key 1`] = `
Object {
  "selectedIndex": -1,
  "showSuggestionList": false,
  "suggestedColors": Array [
    Object {
      "hex": "#f00",
      "name": "red",
    },
    Object {
      "hex": "#f96063",
      "name": "lightred",
    },
  ],
  "value": "red",
}
`;

exports[`color-picker selects color after click on enter key 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="red"
    />
  </div>
</CustomColorPicker>
`;

exports[`color-picker selects color after click on enter key 3`] = `
Object {
  "hex": "#f00",
  "name": "red",
}
`;

exports[`color-picker selects color after click on suggested option 1`] = `
Object {
  "selectedIndex": -1,
  "showSuggestionList": false,
  "suggestedColors": Array [
    Object {
      "hex": "#f96063",
      "name": "lightred",
    },
  ],
  "value": "lightred",
}
`;

exports[`color-picker selects color after click on suggested option 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="lightred"
    />
  </div>
</CustomColorPicker>
`;

exports[`color-picker selects color after click on suggested option 3`] = `
Object {
  "hex": "#f96063",
  "name": "lightred",
}
`;

exports[`color-picker selects color after true name of the color was typed 1`] = `
Object {
  "selectedIndex": -1,
  "showSuggestionList": true,
  "suggestedColors": Array [
    Object {
      "hex": "#f00",
      "name": "red",
    },
    Object {
      "hex": "#f96063",
      "name": "lightred",
    },
  ],
  "value": "red",
}
`;

exports[`color-picker selects color after true name of the color was typed 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="red"
    />
    <ul
      className="suggestions-list"
      onMouseLeave={[Function]}
    >
      <li
        key="red"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f00",
            }
          }
        />
        <Highlight
          str="red"
          substr="red"
        >
          <span>
            <b>
              red
            </b>
          </span>
        </Highlight>
      </li>
      <li
        key="lightred"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f96063",
            }
          }
        />
        <Highlight
          str="lightred"
          substr="red"
        >
          <span>
            light
            <b>
              red
            </b>
          </span>
        </Highlight>
      </li>
    </ul>
  </div>
</CustomColorPicker>
`;

exports[`color-picker selects color after true name of the color was typed 3`] = `
Object {
  "hex": "#f00",
  "name": "red",
}
`;

exports[`color-picker shows list 2nd symbol typed 1`] = `
Object {
  "selectedIndex": -1,
  "showSuggestionList": true,
  "suggestedColors": Array [
    Object {
      "hex": "#f00",
      "name": "red",
    },
    Object {
      "hex": "#f96063",
      "name": "lightred",
    },
  ],
  "value": "re",
}
`;

exports[`color-picker shows list 2nd symbol typed 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="re"
    />
    <ul
      className="suggestions-list"
      onMouseLeave={[Function]}
    >
      <li
        key="red"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f00",
            }
          }
        />
        <Highlight
          str="red"
          substr="re"
        >
          <span>
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
      <li
        key="lightred"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "##f96063",
            }
          }
        />
        <Highlight
          str="lightred"
          substr="re"
        >
          <span>
            light
            <b>
              re
            </b>
            d
          </span>
        </Highlight>
      </li>
    </ul>
  </div>
</CustomColorPicker>
`;

exports[`color-picker shows list 2nd symbol typed 3`] = `undefined`;

exports[`color-picker works after 1st symbol typed 1`] = `
Object {
  "selectedIndex": -1,
  "showSuggestionList": true,
  "suggestedColors": Array [
    Object {
      "hex": "#5f4611",
      "name": "brown",
    },
    Object {
      "hex": "#f00",
      "name": "red",
    },
    Object {
      "hex": "#f96063",
      "name": "lightred",
    },
  ],
  "value": "r",
}
`;

exports[`color-picker works after 1st symbol typed 2`] = `
<CustomColorPicker
  colors={
    Array [
      Object {
        "hex": "#000",
        "name": "black",
      },
      Object {
        "hex": "#5f4611",
        "name": "brown",
      },
      Object {
        "hex": "#f00",
        "name": "red",
      },
      Object {
        "hex": "#f96063",
        "name": "lightred",
      },
      Object {
        "hex": "#fff",
        "name": "white",
      },
    ]
  }
  onChange={[Function]}
>
  <div
    className="custom-color-picker"
  >
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="r"
    />
  </div>
</CustomColorPicker>
`;

exports[`color-picker works after 1st symbol typed 3`] = `undefined`;
